src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -74,9 +74,20 @@ std::optional<GUID> CascadiaSettings::FindGuid(const std::wstring_view profileNa
 {
     std::optional<GUID> profileGuid{};
 
+    std::wstring needle{ profileName };
+    std::transform(needle.begin(),
+                   needle.end(),
+                   needle.begin(),
+                   std::towlower);

```

Comment:
C's tolower() is dependent on the C library locale. Hence, it might not do what the user expects, especially in cases where the Windows language settings are for German or Turkish (sharp S in German, dotted and dotless I as in Eye in Turkish).

Also, any character-by-character case folding may not work for the German sharp S (which is lowercase only, and is sometimes converted to a double regular S on uppercasing; with the opposite conversion sometimes collapsing a double S to a sharp S), among other possible cases.

There's a page on docs.microsoft.com (https://docs.microsoft.com/en-us/windows/win32/intl/handling-sorting-in-your-applications ) that goes into some of the subtleties here.